From 9a9235a8c736a193d656c5406ccb56d0a86ec7fe Mon Sep 17 00:00:00 2001
From: Devi Priya <quic_devipriy@quicinc.com>
Date: Sun, 4 Aug 2024 00:44:51 +0530
Subject: [PATCH] drivers: soc: qcom: Update the firmware sequence for UART SEs

Update the GENI_CFG_RAMn configuration values, index and value offsets
corresponding to GENI_FW_CFG registers for UART SEs.

Change-Id: Iebfccdd5b635f942b106cbc0b85a300a582ab27b
Signed-off-by: Devi Priya <quic_devipriy@quicinc.com>
---
 drivers/soc/qcom/qcom-geni-se.c | 98 ++++++++++++++++-----------------
 1 file changed, 49 insertions(+), 49 deletions(-)

diff --git a/drivers/soc/qcom/qcom-geni-se.c b/drivers/soc/qcom/qcom-geni-se.c
index add4f3f9ed88..fd271fc10a89 100644
--- a/drivers/soc/qcom/qcom-geni-se.c
+++ b/drivers/soc/qcom/qcom-geni-se.c
@@ -933,8 +933,8 @@ typedef struct __attribute__ ((packed)) se_fw_hdr
 } se_fw_hdr;
 
 static uint8_t uart_fw[] __attribute__ ((aligned (4))) = {
-    0x53, 0x45, 0x46, 0x57, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x30, 0x02, 0x00, 0x04, 0x00,
-    0x0a, 0x00, 0xc6, 0x01, 0x1c, 0x00, 0x45, 0x00, 0x48, 0x08, 0x34, 0x07, 0x32, 0x00, 0x00, 0x00,
+    0x53, 0x45, 0x46, 0x57, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x30, 0x02, 0x00, 0x05, 0x00,
+    0x0a, 0x00, 0xc4, 0x01, 0x1c, 0x00, 0x45, 0x00, 0x40, 0x08, 0x2c, 0x07, 0x32, 0x00, 0x00, 0x00,
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x28, 0x0f, 0x00,
     0x20, 0x00, 0x78, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x42, 0x00, 0xc8, 0xc6, 0x7f, 0x00, 0x00, 0x00, 0x3f, 0x00,
@@ -975,17 +975,17 @@ static uint8_t uart_fw[] __attribute__ ((aligned (4))) = {
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x22, 0x00, 0x00, 0x00,
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xc8, 0x0c, 0x00,
     0x10, 0x00, 0x5e, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x93, 0x00,
-    0x08, 0x80, 0x95, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x92, 0x00,
+    0x08, 0x80, 0x94, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
     0x01, 0x7a, 0x7f, 0x00, 0x08, 0x4c, 0x7f, 0x00, 0x01, 0x40, 0x7f, 0x00, 0xf7, 0x79, 0x9d, 0x00,
     0xe9, 0x79, 0x9d, 0x00, 0x41, 0x88, 0x98, 0x00, 0x00, 0x00, 0x0e, 0x00, 0x01, 0x50, 0x7f, 0x00,
-    0x05, 0x20, 0x7f, 0x00, 0x04, 0x19, 0x9f, 0x00, 0x31, 0x8a, 0x98, 0x00, 0xd3, 0x49, 0x9f, 0x00,
-    0x01, 0x44, 0x6a, 0x00, 0x00, 0x80, 0x0b, 0x00, 0x48, 0xc2, 0x98, 0x00, 0x59, 0xb0, 0x98, 0x00,
-    0x43, 0x38, 0x7f, 0x00, 0x03, 0x20, 0x7f, 0x00, 0x04, 0x20, 0x7f, 0x00, 0x66, 0x40, 0x98, 0x00,
+    0x05, 0x20, 0x7f, 0x00, 0x03, 0x20, 0x7f, 0x00, 0x04, 0x19, 0x9f, 0x00, 0x31, 0x8a, 0x98, 0x00,
+    0xd3, 0x49, 0x9f, 0x00, 0x01, 0x44, 0x6a, 0x00, 0x00, 0x80, 0x0b, 0x00, 0x66, 0xc0, 0x98, 0x00,
+    0x58, 0x40, 0x98, 0x00, 0x43, 0x38, 0x7f, 0x00, 0x03, 0x20, 0x7f, 0x00, 0x67, 0x30, 0x9f, 0x00,
     0x62, 0x40, 0x99, 0x00, 0x08, 0x1c, 0x7f, 0x00, 0x10, 0x1c, 0x7f, 0x00, 0x67, 0x78, 0x9f, 0x00,
-    0x00, 0x1e, 0x7f, 0x00, 0x03, 0x20, 0x7f, 0x00, 0x6a, 0x38, 0x99, 0x00, 0x04, 0x1c, 0x7f, 0x00,
+    0x00, 0x1e, 0x7f, 0x00, 0x04, 0x20, 0x7f, 0x00, 0x6a, 0x38, 0x99, 0x00, 0x04, 0x1c, 0x7f, 0x00,
     0x6b, 0x68, 0x9f, 0x00, 0x07, 0x39, 0x68, 0x00, 0x20, 0x38, 0x7f, 0x00, 0x82, 0xb8, 0x7f, 0x00,
     0x91, 0x50, 0x9f, 0x00, 0x1d, 0x41, 0x9f, 0x00, 0x05, 0x39, 0x7f, 0x00, 0x57, 0x11, 0x9f, 0x00,
     0x7b, 0x60, 0x9f, 0x00, 0x01, 0x50, 0x7f, 0x00, 0x9d, 0x48, 0x8b, 0x00, 0x07, 0x39, 0x7f, 0x00,
@@ -996,18 +996,18 @@ static uint8_t uart_fw[] __attribute__ ((aligned (4))) = {
     0x02, 0x60, 0x7f, 0x00, 0xbc, 0x10, 0x9f, 0x00, 0x02, 0x60, 0x7f, 0x00, 0x34, 0x4a, 0x9f, 0x00,
     0xbd, 0x90, 0x98, 0x00, 0xb1, 0x38, 0x9f, 0x00, 0x80, 0x1c, 0x7f, 0x00, 0xbc, 0x00, 0x9a, 0x00,
     0xbc, 0x28, 0x9f, 0x00, 0xb9, 0x18, 0x9f, 0x00, 0x08, 0x60, 0x7f, 0x00, 0x01, 0x20, 0x7f, 0x00,
-    0x10, 0x24, 0x7f, 0x00, 0x4a, 0x48, 0x9f, 0x00, 0xc2, 0x38, 0x99, 0x00, 0x04, 0x1c, 0x7f, 0x00,
-    0x01, 0x40, 0x7f, 0x00, 0xcb, 0x58, 0x9f, 0x00, 0x4a, 0x48, 0x9f, 0x00, 0x45, 0x78, 0x9f, 0x00,
-    0x0a, 0x04, 0x7f, 0x00, 0x4a, 0xc8, 0x9f, 0x00, 0xd2, 0xb8, 0x99, 0x00, 0x04, 0x9c, 0x7f, 0x00,
-    0xf4, 0x98, 0x9f, 0x00, 0xd9, 0x88, 0x98, 0x00, 0x13, 0xf9, 0x9f, 0x00, 0x4a, 0xc8, 0x9f, 0x00,
-    0xd9, 0xd0, 0x9f, 0x00, 0x4a, 0x48, 0x9f, 0x00, 0x02, 0x1c, 0x7f, 0x00, 0xcb, 0x58, 0x9f, 0x00,
-    0xe8, 0x18, 0x9f, 0x00, 0x4a, 0x48, 0x9f, 0x00, 0xcb, 0x78, 0x9f, 0x00, 0x01, 0x1c, 0x7f, 0x00,
-    0x07, 0x20, 0x7f, 0x00, 0x10, 0x60, 0x7f, 0x00, 0x02, 0x20, 0x7f, 0x00, 0x4a, 0x48, 0x9f, 0x00,
-    0x45, 0x78, 0x9f, 0x00, 0x00, 0xa1, 0x7f, 0x00, 0x00, 0x80, 0x1f, 0x00, 0x00, 0xa2, 0x7f, 0x00,
-    0xd4, 0x98, 0x9f, 0x00, 0x10, 0xe0, 0x7f, 0x00, 0x02, 0xa0, 0x7f, 0x00, 0x4a, 0xc8, 0x9f, 0x00,
-    0x45, 0xf8, 0x9f, 0x00, 0x00, 0x21, 0x7f, 0x00, 0x00, 0x00, 0x1f, 0x00, 0x00, 0x22, 0x7f, 0x00,
-    0x46, 0x18, 0x9f, 0x00, 0x10, 0x60, 0x7f, 0x00, 0x02, 0x20, 0x7f, 0x00, 0x47, 0x78, 0x9f, 0x00,
-    0x18, 0x89, 0x9f, 0x00, 0x4a, 0xc8, 0x8e, 0x00, 0xd9, 0xf8, 0x9f, 0x00, 0x4a, 0xc8, 0x8d, 0x00,
+    0x10, 0x24, 0x7f, 0x00, 0x4c, 0x48, 0x9f, 0x00, 0xc2, 0x38, 0x99, 0x00, 0x04, 0x1c, 0x7f, 0x00,
+    0x01, 0x40, 0x7f, 0x00, 0xcb, 0x58, 0x9f, 0x00, 0x4c, 0x48, 0x9f, 0x00, 0x47, 0x78, 0x9f, 0x00,
+    0x0a, 0x04, 0x7f, 0x00, 0x4c, 0xc8, 0x9f, 0x00, 0xd2, 0xb8, 0x99, 0x00, 0x04, 0x9c, 0x7f, 0x00,
+    0xf4, 0x98, 0x9f, 0x00, 0xd9, 0x88, 0x98, 0x00, 0x13, 0xf9, 0x9f, 0x00, 0x4c, 0xc8, 0x9f, 0x00,
+    0xd9, 0xd0, 0x9f, 0x00, 0x4c, 0x48, 0x9f, 0x00, 0x02, 0x1c, 0x7f, 0x00, 0xcb, 0x58, 0x9f, 0x00,
+    0xe8, 0x18, 0x9f, 0x00, 0x4c, 0x48, 0x9f, 0x00, 0xcb, 0x78, 0x9f, 0x00, 0x01, 0x1c, 0x7f, 0x00,
+    0x07, 0x20, 0x7f, 0x00, 0x10, 0x60, 0x7f, 0x00, 0x02, 0x20, 0x7f, 0x00, 0x4c, 0x48, 0x9f, 0x00,
+    0x47, 0x78, 0x9f, 0x00, 0x00, 0xa1, 0x7f, 0x00, 0x00, 0x80, 0x1f, 0x00, 0x00, 0xa2, 0x7f, 0x00,
+    0xd4, 0x98, 0x9f, 0x00, 0x10, 0xe0, 0x7f, 0x00, 0x02, 0xa0, 0x7f, 0x00, 0x4c, 0xc8, 0x9f, 0x00,
+    0x47, 0xf8, 0x9f, 0x00, 0x00, 0x21, 0x7f, 0x00, 0x00, 0x00, 0x1f, 0x00, 0x00, 0x22, 0x7f, 0x00,
+    0x48, 0x18, 0x9f, 0x00, 0x10, 0x60, 0x7f, 0x00, 0x02, 0x20, 0x7f, 0x00, 0x49, 0x78, 0x9f, 0x00,
+    0x18, 0x89, 0x9f, 0x00, 0x4c, 0xc8, 0x8e, 0x00, 0xd9, 0xf8, 0x9f, 0x00, 0x4c, 0xc8, 0x8d, 0x00,
     0xd9, 0xf8, 0x9f, 0x00, 0x39, 0x11, 0x9f, 0x00, 0x1f, 0x61, 0x9f, 0x00, 0x26, 0x21, 0x8b, 0x00,
     0x05, 0x39, 0x7f, 0x00, 0x7b, 0x78, 0x9f, 0x00, 0x05, 0x39, 0x7f, 0x00, 0x2c, 0x21, 0x9a, 0x00,
     0x48, 0x4c, 0x7f, 0x00, 0x31, 0x19, 0x9f, 0x00, 0x08, 0x60, 0x7f, 0x00, 0x01, 0x20, 0x7f, 0x00,
@@ -1031,8 +1031,8 @@ static uint8_t uart_fw[] __attribute__ ((aligned (4))) = {
     0x02, 0x60, 0x7f, 0x00, 0xc0, 0x11, 0x9f, 0x00, 0x02, 0x60, 0x7f, 0x00, 0x01, 0x40, 0x7f, 0x00,
     0xc7, 0x49, 0x9f, 0x00, 0xc3, 0x79, 0x9f, 0x00, 0x08, 0x24, 0x7f, 0x00, 0x01, 0x50, 0x7f, 0x00,
     0xe9, 0x79, 0x9d, 0x00, 0x9f, 0x78, 0x9f, 0x00, 0x02, 0x34, 0x7f, 0x00, 0xb7, 0x79, 0x9f, 0x00,
-    0x01, 0x50, 0x7f, 0x00, 0xda, 0x09, 0x9f, 0x00, 0x45, 0x48, 0x8e, 0x00, 0x4b, 0x78, 0x9f, 0x00,
-    0x45, 0x48, 0x8d, 0x00, 0x4b, 0x78, 0x9f, 0x00, 0xe4, 0x09, 0x9f, 0x00, 0x00, 0x00, 0x0e, 0x00,
+    0x01, 0x50, 0x7f, 0x00, 0xda, 0x09, 0x9f, 0x00, 0x47, 0x48, 0x8e, 0x00, 0x4d, 0x78, 0x9f, 0x00,
+    0x47, 0x48, 0x8d, 0x00, 0x4d, 0x78, 0x9f, 0x00, 0xe4, 0x09, 0x9f, 0x00, 0x00, 0x00, 0x0e, 0x00,
     0x01, 0x00, 0x1f, 0x00, 0x00, 0x00, 0x0d, 0x00, 0x01, 0x00, 0x1f, 0x00, 0x8b, 0x68, 0x7f, 0x00,
     0x40, 0x68, 0x7f, 0x00, 0xf3, 0x41, 0x9f, 0x00, 0xc1, 0x68, 0x7f, 0x00, 0x01, 0x00, 0x1f, 0x00,
     0xc2, 0x68, 0x7f, 0x00, 0x01, 0x00, 0x1f, 0x00, 0x01, 0x2c, 0x7f, 0x00, 0xfd, 0x41, 0x9f, 0x00,
@@ -1042,34 +1042,34 @@ static uint8_t uart_fw[] __attribute__ ((aligned (4))) = {
     0x17, 0x4a, 0x99, 0x00, 0x2b, 0x4a, 0x9f, 0x00, 0x1d, 0x0a, 0x9f, 0x00, 0x13, 0x7a, 0x9f, 0x00,
     0x2b, 0x4a, 0x9f, 0x00, 0x80, 0x36, 0x7f, 0x00, 0x80, 0x35, 0x7f, 0x00, 0x2b, 0x7a, 0x9f, 0x00,
     0x05, 0x39, 0x7f, 0x00, 0x01, 0x34, 0x7f, 0x00, 0x01, 0x34, 0x7f, 0x00, 0x03, 0x7a, 0x9f, 0x00,
-    0x2b, 0x6a, 0x9f, 0x00, 0x01, 0x50, 0x7f, 0x00, 0x9f, 0x78, 0x9f, 0x00, 0x45, 0x48, 0x9f, 0x00,
-    0x4b, 0x78, 0x9f, 0x00, 0x4b, 0x90, 0x98, 0x00, 0x3b, 0x3a, 0x9f, 0x00, 0x80, 0x1c, 0x7f, 0x00,
-    0x4a, 0x00, 0x9a, 0x00, 0x4a, 0x28, 0x9f, 0x00, 0x43, 0x1a, 0x9f, 0x00, 0x08, 0x60, 0x7f, 0x00,
-    0x01, 0x20, 0x7f, 0x00, 0x10, 0x24, 0x7f, 0x00, 0x4b, 0x78, 0x9f, 0x00, 0x05, 0x20, 0x7f, 0x00,
-    0x67, 0x78, 0x9f, 0x00, 0x51, 0x8a, 0x98, 0x00, 0x00, 0x00, 0x0d, 0x00, 0x57, 0x08, 0x7f, 0x00,
-    0x29, 0x10, 0x7f, 0x00, 0x59, 0x02, 0x9f, 0x00, 0x00, 0x40, 0x1f, 0x00, 0x01, 0x60, 0x7f, 0x00,
-    0x57, 0x7a, 0x9f, 0x00, 0x24, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x9a, 0x01, 0x00, 0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0xe0, 0x00,
-    0x20, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x00, 0x00,
-    0x00, 0x07, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x00, 0xc0, 0x00, 0x00, 0x0b, 0x00, 0x00,
-    0x31, 0x02, 0x02, 0x00, 0x05, 0xce, 0x00, 0x00, 0xe7, 0x60, 0x03, 0x00, 0xa8, 0xe6, 0x41, 0x09,
-    0x10, 0x05, 0x10, 0x00, 0x51, 0x1e, 0xc0, 0x42, 0x01, 0x04, 0x00, 0x00, 0x00, 0x84, 0x2e, 0x00,
-    0x1a, 0x58, 0x94, 0x16, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
-    0x1c, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x2b, 0x6a, 0x9f, 0x00, 0x01, 0x50, 0x7f, 0x00, 0x9f, 0x78, 0x9f, 0x00, 0x47, 0x48, 0x9f, 0x00,
+    0x4d, 0x78, 0x9f, 0x00, 0x4d, 0x90, 0x98, 0x00, 0x3b, 0x3a, 0x9f, 0x00, 0x80, 0x1c, 0x7f, 0x00,
+    0x4c, 0x00, 0x9a, 0x00, 0x4c, 0x28, 0x9f, 0x00, 0x43, 0x1a, 0x9f, 0x00, 0x08, 0x60, 0x7f, 0x00,
+    0x01, 0x20, 0x7f, 0x00, 0x10, 0x24, 0x7f, 0x00, 0x4d, 0x78, 0x9f, 0x00, 0x4d, 0x8a, 0x98, 0x00,
+    0x00, 0x00, 0x0d, 0x00, 0x57, 0x08, 0x7f, 0x00, 0x29, 0x10, 0x7f, 0x00, 0x55, 0x02, 0x9f, 0x00,
+    0x00, 0x40, 0x1f, 0x00, 0x01, 0x60, 0x7f, 0x00, 0x53, 0x7a, 0x9f, 0x00, 0x24, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x24, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x9a, 0x01, 0x00,
+    0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0xe0, 0x00, 0x20, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x07, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x24, 0x00, 0xc0, 0x00, 0x00, 0x0b, 0x00, 0x00, 0x31, 0x02, 0x02, 0x00, 0x05, 0xce, 0x00, 0x00,
+    0xe7, 0x60, 0x03, 0x00, 0xa8, 0xe6, 0x41, 0x09, 0x10, 0x05, 0x10, 0x00, 0x51, 0x1e, 0xc0, 0x42,
+    0x01, 0x04, 0x00, 0x00, 0x00, 0x84, 0x2e, 0x00, 0x1a, 0x58, 0x94, 0x16, 0x00, 0x00, 0x00, 0x00,
     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x06, 0x1c, 0x08, 0x00, 0x10, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0d, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x06, 0x1c, 0x08, 0x00, 0x10, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x15, 0x24, 0xc0, 0x00,
-    0x0e, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x55, 0x00, 0x00, 0x00, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
-    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10, 0x11, 0x12, 0x40, 0x41, 0x42, 0x43, 0x44,
-    0x45, 0x46, 0x47, 0x48, 0x49, 0x4a, 0x4b, 0x4c, 0x4d, 0x4e, 0x4f, 0x50, 0x51, 0x52, 0x53, 0x54,
-    0x55, 0x56, 0x57, 0x58, 0x59, 0x5a, 0x5b, 0x5c, 0x5d, 0x5e, 0x5f, 0x60, 0x61, 0x62, 0x63, 0x64,
-    0x65, 0x66, 0x67, 0x68, 0x69, 0x6a, 0x6b, 0x6c, 0x6d, 0x6e, 0x6f, 0x70, 0x71
+    0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x1c, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x0f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06, 0x1c, 0x08, 0x00, 0x10, 0x40, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x0d, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06, 0x1c, 0x08, 0x00,
+    0x10, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x15, 0x24, 0xc0, 0x00, 0x0e, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
+    0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x55, 0x00, 0x00, 0x00,
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+    0x10, 0x11, 0x12, 0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48, 0x49, 0x4a, 0x4b, 0x4c,
+    0x4d, 0x4e, 0x4f, 0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57, 0x58, 0x59, 0x5a, 0x5b, 0x5c,
+    0x5d, 0x5e, 0x5f, 0x60, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6a, 0x6b, 0x6c,
+    0x6d, 0x6e, 0x6f, 0x70, 0x71
 };
 
 static uint8_t spi_fw[] __attribute__ ((aligned (4))) = {
-- 
2.34.1

